compiler/noirc_evaluator/src/ssa/opt/defunctionalize.rs

```diff
@@ -278,8 +282,10 @@ fn create_apply_function(
     function_ids: Vec<FunctionId>,
 ) -> FunctionId {
     assert!(!function_ids.is_empty());
-    ssa.add_fn(|id| {
+    let globals_dfg = std::mem::take(&mut ssa.globals.dfg);
+    let new_id = ssa.add_fn(|id| {
         let mut function_builder = FunctionBuilder::new("apply".to_string(), id);
+        function_builder.set_globals(Arc::new(GlobalsGraph::from_dfg(globals_dfg.clone())));
```

Comment:
I don't understand why we do the whole `std::mem::take` thing with the globals dfg here. Shouldn't we have a `Arc<GlobalsGraph>` somewhere? If `Ssa` doesn't already store a `Arc<GlobalsGraph>` then it probably should - we should avoid cloning entire DFGs if possible - even small ones like globals is likely expected to be.

Reply:
I switched to using `function_ids` to access a function's globals graph. This is safe due to `assert!(!function_ids.is_empty());`.